src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -1134,29 +1135,46 @@ static string GetSchemaOrDefault([CanBeNull] string schema, [CanBeNull] IAnnotat
         bool VersionAtLeast(int major, int minor)
             => _postgresVersion is null || new Version(major, minor) <= _postgresVersion;
 
-        string IndexColumnList(string[] columns, string[] operators)
+        string IndexColumnList(string[] columns, string[] operators, string[] sortOptions)
         {
-            if (operators == null || operators.Length == 0)
-                return ColumnList(columns);
+            var isFirst = true;
+            var builder = new StringBuilder();
 
-            return string.Join(", ", columns.Select((v, i) =>
+            for (int i = 0; i < columns.Length; i++)
             {
-                var identifier = Dependencies.SqlGenerationHelper.DelimitIdentifier(v);
+                if (!isFirst)
+                    builder.Append(", ");
 
-                if (i >= operators.Length)
-                    return identifier;
+                builder.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(columns[i]));
 
-                var @operator = operators[i];
+                if (operators != null && i < operators.Length)
+                {
+                    var @operator = operators[i];
 
-                if (string.IsNullOrEmpty(@operator))
-                    return identifier;
+                    if (!string.IsNullOrEmpty(@operator))
+                    {
+                        var delimitedOperator = TryParseSchema(@operator, out var name, out var schema)
+                            ? Dependencies.SqlGenerationHelper.DelimitIdentifier(name, schema)
+                            : Dependencies.SqlGenerationHelper.DelimitIdentifier(@operator);
 
-                var delimitedOperator = TryParseSchema(@operator, out var name, out var schema)
-                    ? Dependencies.SqlGenerationHelper.DelimitIdentifier(name, schema)
-                    : Dependencies.SqlGenerationHelper.DelimitIdentifier(@operator);
+                        builder.Append(" ").Append(delimitedOperator);
+                    }
+                }
+
+                if (sortOptions != null && i < sortOptions.Length)
+                {
+                    var sortOption = sortOptions[i];
+
+                    if (!string.IsNullOrEmpty(sortOption))
+                    {
+                        builder.Append(" ").Append(sortOption);
```

Comment:
If we don't validate `sortOption` above, should we delimit it here?

Reply:
What happens if you delimit some of these (valid) values: `COLLATE`, `ASC`, `DESC`, `NULLS FIRST` and `NULLS LAST`? 🤔